include/aws/s3/private/s3_request.h

```diff
@@ -158,6 +158,8 @@ struct aws_s3_request_metrics {
         struct aws_string *ip_address;
         /* The pointer to the connection that request was made from */
         void *connection_id;
+        /* The pointer to the request that the request attempt was made from */
+        void *request_id;
```

Comment:
let's keep this as our internal, ptr or something to be different from the request id for the server.